### ISSUE TITLE
Unable to parse MySQL multitable delete

### ISSUE THREAD

**Issue by jasonaowen** [sqlfluff/sqlfluff#2819]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff is unable to parse MySQL's [multitable delete syntax](https://dev.mysql.com/doc/refman/5.7/en/delete.html#idm45288826299904).

### Expected Behaviour

SQLFluff to parse the valid, executable SQL and report on any rule violations.

### Observed Behaviour

Instead, SQLFluff reports:

> Line 1, Position 1: Found unparsable section: 'DELETE a FROM a JOIN b USING (id) WHERE ...'

### How to reproduce

With `test.sql` as:

```sql
DELETE a FROM a JOIN b USING (id) WHERE b.name = 'example';
```

running SQLFluff results in a parse error:

```sh
$ sqlfluff lint --verbose --ignore-local-config --dialect mysql test.sql 
==== sqlfluff ====
sqlfluff:               0.11.0 python:                 3.8.10
implementation:        cpython dialect:                 mysql
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: test.sql ] ===

== [test.sql] FAIL                                                                                                                                                       
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'DELETE a
                       | FROM a JOIN b USING (id) WHERE ...'
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

For reference, here's a schema and example data for how multitable deletion works:

```sql
CREATE TABLE a (id INTEGER PRIMARY KEY);
CREATE TABLE b (id INTEGER PRIMARY KEY, name TEXT NOT NULL);
INSERT INTO a (id) VALUES (1), (2);
INSERT INTO b (id, name) VALUES (1, 'one'), (2, 'example');
```

Running the example SQL should result in table `a` having a single row, `1`, and table `b` still having both rows.

### Dialect

MySQL

### Version

sqlfluff, version 0.11.0
Python 3.8.10

### Configuration

All defaults

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
This turned out to be trickier than I thought it was going to be! Anyway think I finally got there with #2823

**Comment by jasonaowen** [comment]
Thanks for the quick fix, @tunetheweb, and the quick review by @barrywhart!
